packages/mdc-list/foundation.js

```diff
@@ -109,29 +112,36 @@ class MDCListFoundation extends MDCFoundation {
 
   /** @param {number} index */
   setSelectedIndex(index) {
-    if (index === this.selectedIndex_) {
+    if (index < 0 && index >= this.adapter_.getListItemCount()) {
```

Comment:
Why does this no longer bail out if `setSelectedIndex` is called redundantly?

Also, I suspect we intentionally did not bail out immediately if selectedIndex was out of range, in order to support setting it to -1. @williamernest is that correct?

Reply:
Didn't see any documentation related to negative tab index deselects the list.

Regarding redundantly setting the `setSelectedIndex` on same index - it actually toggles the checkbox if called twice on same index.

Reply:
Discussed w/ Abhinay; this logic should be reasonable for this PR, since `setSelectedIndex` formerly didn't do anything for checkboxes at all.

It also turns out that we already worded this as "Toggles" in the readme, which is more correct now than it was before. We might want to clarify that it toggles for checkboxes but always sets the indicated index as selected for radios and single selection.